src/contracts/facilitators/gsm/GhoReserve.sol

```diff
@@ -47,6 +47,8 @@ contract GhoReserve is Ownable, VersionedInitializable, IGhoReserve {
 
   /// @inheritdoc IGhoReserve
   function use(uint256 amount) external {
+    require(_entities.contains(msg.sender), "NOT_ENTITY");
```

Comment:
is this check actually needed? same in `restore`
- when removing the entity, we check used and limit is 0.
- when adding an entity, it can start with 0 limit.
- use check limit, so only existent entities with non-zero limit are allowed
- restore check used, so only existent entities with non-zero used are allowed

Reply:
apart from this, we are doing a similar check in `setLimit` with different error. We should be consistent
```
  function setLimit(address entity, uint256 limit) external onlyOwner {
    require(_entities.contains(entity), 'ENTITY_DOES_NOT_EXIST');
```